docs/wiki/Vision.md

```diff
@@ -0,0 +1,115 @@
+- The fundamental **modules** that each deploy a service, i.e. are the building blocks to deploy environments
+
+To make each component a bit more tangible, let's take a look at the following example:
+
+- Target Environment
+  - A virtual machine connected to a storage account
+- Orchestration
+  - Template-orchestration using GitHub actions
+- Modules
+  - Resource Group
+  - Virtual Network
+  - Virtual Machine
+  - Storage Account
+
+For this example we could create a GitHub workflow that signs into Azure and run's a single deployment using for example the PowerShell command `New-AzResourceGroupDeployment`.
+Furthermore we'd then create an orchestration-template the deploys the above resources in the following parallel groups (using dependencies)
+1. The resource group
+1. The virtual network & storage account
+1. The virtual Machine
+
+Then we'd only need to create a parameter file for the orchestration-template and have the workflow deploy both in combination.
+
+### What is the intended deployment flow?
+
+In this section we'll take a deeper look into the fundamental flow of the platform.
+
+First things first, we would work towards the deployment of our environments in 3 phases:
+
+<img src="media/deploymentFlow.png" alt="Deployment flow" height="150">
+
+The **Produce components** phase is the phase where you're supposed to set up your modules and validate that they work. In case of _CARML_ this means we have a pipeline for each module that, with one or multiple test-parameter files, runs static as well as validation & deployment tests on its template. Once if all tests pass we can be certain no bugs were introduced (in case the template was altered) and all features work as intended.
+
+The next phase, **Publish artifacts**, will take the tested and approved modules and publish them to a target location of your choice (for example _template specs_ or the _bicep registry_). The publishing should publish at least the tested module template itself.
+The target location should support versioning so that you only always publish new versions.
+This may bring up the question _Why_ you should publish your templates before you use them. Afterall, you could just reference the modules in the source repository as is to deploy you environments. This however has one major drawback: If your deployments rely on what you have in your source repository then they 'by definition' always use the latest code. This means, as soon as somebody introduces a breaking change to one of these templates, all your orchestrated and referencing deployments will be blocked until you updated them to the latest version too.
```

Comment:
This deserves its own paragraph and header

Reply:
Moved it down to `Why use versioned modules?`

Reply:
Yes, that's much better. Quick comment: '**The excellent question**', doesn't fit here. Maybe something like this: </br>
Today, the customary practice for deploying resources from modules is to reference them from source control. This, however, has one major drawback: If your deployments rely on what you have in your source repository then 'by definition' will use the latest code in source control. Applying software development lifecycle concepts like 'publishing build artifacts and versioning' enables you to have a point in time version of an Azure Resource Module. By introducing versions to your modules, the consuming orchestration can and should specify a module version it wants to use and deploy the Azure environment using them. If we now have the case that a breaking change is introduced and an updated version is published, no deployments are affected because they still reference the previously published version. Instead, they must make the deliberate decision to upgrade the module to reference newer versions.

Reply:
actually I doubt it is a common practice to reference resources from source control. However, independent of that fact I try to explain 'why' it would be a bad practice regardless

Reply:
I slightly altered your text to account for that. Please check